Sources/BSON/ExtendedJSONDecoder.swift

```diff
@@ -0,0 +1,23 @@
+import Foundation
+/// `ExtendedJSONDecoder` facilitates the decoding of ExtendedJSON into `Decodable` values.
+public class ExtendedJSONDecoder {
+    /// Initialize an `ExtendedJSONDecoder`.
+    public init() {
+        fatalError("unimplemented")
+    }
+
+    /// Decodes an instance of the requested type `T` from the provided extended JSON data.
+    ///
+    /// - Parameters:
+    ///   - type: Codable type to decode the input into.
+    ///   - data: `Data` which represents the JSON that will be decoded.
+    /// - Returns: Decoded representation of the JSON input as an instance of T.
+    /// - Throws:
```

Comment:
whoops, I missed before that we don't actually have an error here. I think you could probably just copy the equivalent error info from `BSONDecoder`.

Reply:
Yeah I thought I would just fill this in once I actually do the implementation, but I can do this instead